### PR TITLE
Updated setup.sh to account for RPM distros

### LINKED ISSUE #16: Update `setup.sh` script to prefer RPM where available
Our current `setup.sh` script looks for `apt`, and if it's there, installs Enclave from our repository. 

We should do the same thing for RPM-compatible distros.

### PR DESCRIPTION
**Pull request by ghost**
closes #16 

# Tested on
Fedora 36
OpenSuse Leap 15.3
Amazon Linux 2
CentOS 7

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -70,17 +74,60 @@ install_apt_package() {
...
     fi
 
     # Do not continue with rest of setup as deb handles all of it
     exit 0
 }
 
+
+install_yum_package() {
+    # Install the pre-requisites for a yum install
+    info "Installing Pre requisits."
```
Comment (enclave-alistair) [inline setup.sh]:
```suggestion
    info "Installing Prerequisites."
```

**Review by enclave-alistair — CHANGES_REQUESTED** [review]
Looks great, thanks. One typo.

Can you drop a list of the tested distros here?

**Review by enclave-alistair — DISMISSED** [review]
👍

File: setup.sh [hunk 2]
```diff
@@ -217,8 +293,9 @@ ENCLAVE_PKG_LIST="${ENCLAVE_PKG_LIST:-enclave.stable.list}"
 
 # Check if a package is available and install,
 # if no package, install dependencies
-preinstall
-# Install manually (no package available)
+if preinstall; then
 install_enclave
+fi
+# Install manually (no package available)
```
Comment (enclave-alistair) [inline setup.sh]: Comment position?

File: setup.sh [hunk 3]
```diff
@@ -70,17 +92,60 @@ install_apt_package() {
...
+        sudo apt install -yq "enclave=${ENCLAVE_VERSION}"
     else
         # Install latest Enclave
         info "Installing Enclave package (latest)."
-        sudo apt-get install -yq enclave
+        sudo apt install -yq enclave
     fi
 
     # Do not continue with rest of setup as deb handles all of it
     exit 0
```
Comment (enclave-alistair) [inline setup.sh:103]: Do we need to change the apt path to function more like the RPM one now? It used to be that the deb file did the enrolment in-line, but we've stopped that now.

Comment (enclave-alistair) [inline setup.sh]: Do we need to change the apt path to function more like the RPM one now? It used to be that the deb file did the enrolment in-line, but we've stopped that now.  Right now apt bails from the install script after the package install.

**Review by enclave-alistair — COMMENTED** [review]
Query around apt behaviour differing from rpm, but otherwise looking good.

File: setup.sh (hunk 3 above)
Comment (ghost) [reply setup.sh]: Just updated this all sorted!

**Review by enclave-alistair — APPROVED** [review]
Lovely stuff. 🥳

**Pull request merged**